### PR TITLE
Fix add some useful components to empty loans pages

### LINKED ISSUE #5979: My Loans Page: Show useful content if no loans
### Describe the problem that you'd like solved

When a patron has no loans and visits their `/account/loans` page, they are presented with a mostly blank page:

Screenshot from 2021-12-15 16-44-40

This blank page is an opportunity to help patrons find their next book to borrow.

### Proposal & Constraints

If a patron has currently has no loans, let's display the following on the loans page:
- [ ] A "Books We Love" carousel
- [ ] A "Browse by Subject" carousel
- [ ] A link to our collections page (https://openlibrary.org/collections)

There may be some issues with the carousels overflowing on the loans page.  The carousels can be configured to display a smaller number of cards, which may prevent the carousels from overflowing.

The collections page link should have a nice call to action, something like "Check out these special collections".

### Additional context

Loans page template: https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/account/loans.html

Example "Books We Love" carousel call: https://github.com/internetarchive/openlibrary/blob/9dbb9c13cda88c0c79b2fab0f47b64f72d75f681/openlibrary/templates/home/index.html#L27
Example "Browse by Subject" carousel call: https://github.com/internetarchive/openlibrary/blob/9dbb9c13cda88c0c79b2fab0f47b64f72d75f681/openlibrary/templates/home/index.html#L41

### PR DESCRIPTION
**Pull request by tuminzee**
Closes #5979

#### Enhance user experience/engagement  on `/account/loans` page, 
If a patron has currently has no loans, display the following on the loans page:

- "Books We Love" carousel  [because of failure of creating the book we love carousel, instead I have replace it with 'Classic Books Carousel]
- "Browse by Subject" carousel
- link to our collections page (https://openlibrary.org/collections)

### Technical

1. Add boolean paramater <compact_mode> to custom_ia_carousel, custom_carousel template which will determine the breaking behaviour
1.2 Set breaking [a:f] pareters [4, 4, 4, 3, 2, 1]for the compact design
2. Make css changes in mybooks-details.less to make my loan page's carousel's responsive
3. Add action button to my loan's page which redirects to collections page

### Testing

1. Go to /accounts/loan 
2. DevTools Responsive Breakpoints

### Stakeholders

@jimchamp

### CODE REVIEW

**Comment by tuminzee** [comment]
@cclauss I am super happy. I was able to pass the GitHub actions tests.
Although I tested the code locally, I was still unsure.

Thank you for the lesson.
I hope you have a good day 😄

**Comment by tuminzee** [comment]
@jimchamp This issue helped a lot to learn about the codebase. Thank you for assigning this 🕺🏼 
Thank you for being patient with me. Learnt a lot.
Looking forward to contributing more 

Please let me know if you are not satisfied with the PR. I I will make the necessary changes.

File: openlibrary/templates/account/loans.html [hunk 1]
```diff
@@ -55,6 +55,9 @@
 <div>
   $if len(loans) == 0:
     <div><em>$_("You've not checked out any books at this moment.")</em></div>
+    $:render_template("books/custom_carousel", books=readonline_carousel(), title=_('Classic Books'), url="/read", key="public_domain", test=test, compact_mode=True)
```
Comment (jimchamp) [inline openlibrary/templates/account/loans.html:58]:
```suggestion
    <div>Looking for a book to borrow?  Check out our staff picks, or search for a book on a specific category:</div>
     $:render_template("home/custom_ia_carousel", title=_('Books We Love'), key="staff_picks", query='languageSorter:("English")', subject="openlibrary_staff_picks", sorts=["lending___last_browse desc"], limit=18, test=test, compact_mode=True)
```
I think that this is missing a call to action.  Let's add that before any of the carousels.  You may want to add a bit of space between the "You've not checked out any books at this moment." message and the call to action (unless you think that it looks good).

Although we can't easily test the "Books We Love" carousel locally, we can test this by deploying to https://testing.openlibrary.org.  Let's try replacing the "Classic Books" carousel with the "Books We Love" carousel.  Once that change has been made, I'll deploy this branch to testing.

File: openlibrary/templates/account/loans.html [hunk 2]
```diff
@@ -55,6 +55,9 @@
...
+    $:render_template("home/categories", test=test)
+    Check out our <a href="/collections">special collections</a>
```
Comment (jimchamp) [inline openlibrary/templates/account/loans.html:60]:
```suggestion
    Or, check out our <a href="/collections">special collections</a>.
```
Adding "Or" to the beginning of this sentence makes more sense with the call to action.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks for your patience as well, @tuminzee!  Just a few small changes are needed.  Afterwards, I'll put this on testing in order to ensure that the staff favorites carousel is rendering as expected.

**Comment by tuminzee** [comment]
I have made the changes, please have a look.
Thank you for your time 😄

**Comment by jimchamp** [comment]
This is up on testing now, and it's looking pretty good:

Screenshot from 2021-12-23 14-45-17

**Comment by tuminzee** [comment]
This looks great. Thank you for all the help.
I wish this makes it to the main branch 🤞🏼

**Comment by jimchamp** [comment]
I'm going to make a small change to the copy, and merge everything now.

File: openlibrary/templates/account/loans.html [hunk 3]
```diff
@@ -55,6 +55,10 @@
 <div>
   $if len(loans) == 0:
     <div><em>$_("You've not checked out any books at this moment.")</em></div>
+    <div>Looking for a book to borrow?  Check out our staff picks, or search for a book on a specific category:</div>
```
Comment (jimchamp) [inline openlibrary/templates/account/loans.html:58]:
```suggestion
    <br>
    <div>Looking for a book to borrow?  Check out our staff picks, or search for a book on a specific subject:</div>
```

**Review by jimchamp — APPROVED** [review]
Lgtm!  Thanks @tuminzee!

**Comment by tuminzee** [comment]
PR merged 🎉 🕺🏼 
Thank you @jimchamp

**Pull request merged**